src/ml/neural_net/mps_layer_instance_norm_data_loader.mm

```diff
@@ -69,101 +69,143 @@ - (instancetype) initWithParams:(NSString *)name
       ones_ptr[x] = 1.0f;
     }
 
+    _adamGamma = [[MPSNNOptimizerAdam alloc] initWithDevice:dev
+                                              learningRate:0.001f];
+        
+    _adamBeta = [[MPSNNOptimizerAdam alloc] initWithDevice:dev
+                                             learningRate:0.001f];
+
     _vDesc = [MPSVectorDescriptor vectorDescriptorWithLength:_numberOfFeatureChannels
                                                    dataType:(MPSDataTypeFloat32)];
 
-    _gammaBuffer = [dev newBufferWithBytes:_gamma_weights
-                                    length:sizeof(float) * _numberOfFeatureChannels
-                                   options:MTLResourceStorageModeManaged];
+    _gammaVectorArray = [[NSMutableArray alloc] init];
+    _betaVectorArray = [[NSMutableArray alloc] init];
+    _gammaMomentumVectorArray = [[NSMutableArray alloc] init];
+    _gammaVelocityVectorArray = [[NSMutableArray alloc] init];
+    _betaMomentumVectorArray = [[NSMutableArray alloc] init];
+    _betaVelocityVectorArray = [[NSMutableArray alloc] init];
 
-    _betaBuffer = [dev newBufferWithBytes:_beta_weights
-                                   length:sizeof(float) * _numberOfFeatureChannels
-                                  options:MTLResourceStorageModeManaged];
+    _gammaBufferArray = [[NSMutableArray alloc] init];
+    _betaBufferArray = [[NSMutableArray alloc] init];
 
+    _movingMeanBufferArray = [[NSMutableArray alloc] init];
+    _movingVarianceBufferArray = [[NSMutableArray alloc] init];
+    _gammaMomentumBufferArray = [[NSMutableArray alloc] init];
+    _gammaVelocityBufferArray = [[NSMutableArray alloc] init];
+    _betaMomentumBufferArray = [[NSMutableArray alloc] init];
+    _betaVelocityBufferArray = [[NSMutableArray alloc] init];
 
-    _gammaMomentumBuffer = [dev newBufferWithBytes:zeros_ptr
-                                            length:sizeof(float) * _numberOfFeatureChannels
-                                           options:MTLResourceStorageModeManaged];
+    for (size_t index = 0; index < styles; index ++){
```

Comment:
Let's use NSUInteger instead of size_t here, since this is principally a Obj C file

Reply:
Got it!